File: frontend/src/app/shared/models/attributes/spell-school.ts
import {Attribute} from './attribute';
import {AttributeType} from './attribute-type.enum';

export class SpellSchool extends Attribute {
  type = 'SpellSchool';

  constructor() {
    super();
    this.attributeType = AttributeType.SPELL_SCHOOL;
  }
}

File: backend/src/main/java/com/herd/squire/models/attributes/SpellSchool.java
package com.herd.squire.models.attributes;

public class SpellSchool extends Attribute {
    public SpellSchool() {}

    public SpellSchool(String id, String name, String description, int sid, boolean author, int version) {
        super(id, name, description, AttributeType.SPELL_SCHOOL, sid, author, version);
    }
}
